### PR TITLE
MA now has a checkbox for autoscale

### LINKED ISSUE #29065: Replace autoscale in MA with a checkbox
We want to replace the `autoscale` button in the muon analysis GUI with an `autoscale` checkbox. When it is ticked the y scale will always be autscaled. 

It should:
1. Disable the y selection range when ticked
2. Be unticked automatically when the user uses the pan or zoom
3. When ticked autoscale when the data is changed (i.e. load new runs)
4. When ticked autoscale when the data is updated (e.g. update alpha)

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
Wanted to add a checkbox for MA and FDA.
Unit tests will be added in #29321

**To test:**
Open MA/FDA
Load data
Check if the autoscale works (e.g. change data, rebin, change groups/pairs)
Do not use fitting it has a known bug

Open Elemental analysis
Load data
Create a plot
The autoscale should still be a button

Fixes #29065. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_toolbar.py [hunk 1]
```diff
@@ -96,3 +101,77 @@ def show_minor_gridlines(self):
...
+        for axes in self.canvas.figure.get_axes():
+            axes.set_navigate_mode(self._active)
+
+        self.set_message(self.mode)
+
+    def home(self, *args):
+        """Restore the original view."""
+        self._nav_stack.home()
+        self.set_history_buttons()
+        self._update_view()
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_toolbar.py:177]:
```suggestion
        self._update_view()

```

File: scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_presenter.py [hunk 2]
```diff
@@ -26,15 +27,26 @@ def __init__(self, view: PlottingCanvasViewInterface, model: PlottingCanvasModel
...
+    def _setup_quick_edit_widget(self):
         self._options_presenter.connect_errors_changed(self.handle_error_selection_changed)
         self._options_presenter.connect_x_range_changed(self._handle_xlim_changed_in_quick_edit_options)
         self._options_presenter.connect_y_range_changed(self._handle_ylim_changed_in_quick_edit_options)
         self._options_presenter.connect_autoscale_changed(self._handle_autoscale_y_axes)
         self._options_presenter.connect_plot_selection(self._handle_subplot_changed_in_quick_edit_widget)
 
+    def _setup_autoscale_observer(self):
+        self.uncheck_autoscale_observer = GenericObserver(self._options_presenter.uncheck_autoscale)
+        self._view.toolBar.uncheck_autoscale_notifier.add_subscriber(self.uncheck_autoscale_observer)
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_presenter.py:42]:
```suggestion
        self.add_uncheck_autoscale_subscriber(self.uncheck_autoscale_observer)
```

**Comment by AnthonyLim23** [comment]
I have made a new issue to refactor the code https://github.com/mantidproject/mantid/issues/29323

**Review by StephenSmith25 — APPROVED** [review]
Although this currently lacks unit tests, it is a feature that is preventing some of the scientists from using the GUI. Therefore I'm going to approve it. Having discussed this with Anthony, we think this feature (/the widget) will need rethinking come the next maintenance period, where there will be time to think of a more robust (testable) implementation.

Just waiting on those last builds to pass.

**Pull request merged**